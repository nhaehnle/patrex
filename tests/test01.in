boost::bind(&foo, boost::ref(*this));
boost::bind(&foo, a, boost::ref(*this));
boost::bind(&A::foo, boost::ref(*this));
boost::bind(&A::foo, a, boost::ref(*this));
