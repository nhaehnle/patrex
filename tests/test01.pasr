
qname = $(id) :: $(qname)
qname = $(id) :: $(id)
r = boost::bind(& $(qname), $(( boost::ref(*this) ))|ref| $*)

r:
	replace ref "this"
