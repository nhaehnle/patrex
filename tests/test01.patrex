
match boost::bind(& ${id} $( :: ${id} )+, $( boost::ref(*this) )|ref| $* )
	replace ref "this"
