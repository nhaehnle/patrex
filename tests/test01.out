boost::bind(&foo, boost::ref(*this));
boost::bind(&foo, a, boost::ref(*this));
boost::bind(&A::foo, this);
boost::bind(&A::foo, a, boost::ref(*this));
boost::bind(decoy, &A::foo, boost::ref(*this));
